src/main/webapp/availability.js

```diff
@@ -38,3 +38,22 @@ function browserTimezoneOffset() {
 function availabilityTableDiv() {
   return document.getElementById('table-container');
 }
+
+function updateAvailability() {
+  let selectedSlots = document.getElementsByClassName('table-success');
+  let requestBody = [];
+  let firstAndLastSlots = [];
```

Comment:
Nit: You can remove basically all of these variable declarations by using a declarative style.

```javascript
let requestBody = [
  [
    document.getElementById('first-slot').getAttribute('data-utc'),
    document.getElementById('last-slot').getAttribute('data-utc'),
  ],
  selectedSlots.map(s => s.getAttribute('data-utc'))
]
```

After writing that though, i suspect that's not the logic you actually had in mind. Maybe this instead?

```javascript
let request_body = [
  document.getElementById('first-slot').getAttribute('data-utc'),
  document.getElementById('last-slot').getAttribute('data-utc'),
].concat(selectedSlots.map(s => s.getAttribute('data-utc'));
```

My preferred way to implement this would be to collect all the DOM elements and then get the attributes though

```javascript
let request_body = selectedSlots
  .concat([
     document.getElementById('first-slot'),
     document.getElementById('last-slot'),
  ])
  .map(el => el.getAttribute('data-utc'));

Reply:
Just a not: .map does not work on HTMLCollections, used Array.from() to solve that problem.